sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBufferStore.scala

```diff
@@ -50,24 +50,47 @@ abstract class RapidsBufferStore(
     private[this] val comparator: Comparator[RapidsBufferBase] =
       (o1: RapidsBufferBase, o2: RapidsBufferBase) =>
         java.lang.Long.compare(o1.getSpillPriority, o2.getSpillPriority)
+    // buffers: contains all buffers in this store, whether spillable or not
     private[this] val buffers = new java.util.HashMap[RapidsBufferId, RapidsBufferBase]
```

Comment:
This is being modified/read without the store's lock held.  What lock is supposed to be held when doing this?

Reply:
`buffers` is modified with the `BufferTracker` lock held. This line is happening during initialization, which should be happening before tasks start to run. So I think this is safe, but let me know if I've missed something.